kernel/aster-nix/src/fs/utils/inode.rs

```diff
@@ -55,6 +55,18 @@ impl InodeType {
     pub fn is_device(&self) -> bool {
         *self == InodeType::BlockDevice || *self == InodeType::CharDevice
     }
+
+    pub fn from_raw_mode(mut mode: u16) -> Result<Self> {
```

Comment:
This API deserves more document since it interacts with variables from user in our syscall layer, not with our fs subsystem, so there is a gap in terms of readability.